--- commons-daemon-1.0.15-src/src/native/unix/configure	2018-04-17 07:22:48.919068214 +0000
+++ commons-daemon-1.0.15-src/src/native/unix/configure	2018-04-17 07:45:24.880251314 +0000
@@ -2695,6 +2695,11 @@
         HOST_CPU=ia64
     fi
     ;;
+  e2k*)
+    CFLAGS="$CFLAGS -DCPU=\\\"e2k\\\""
+    supported_os="e2k"
+    HOST_CPU=e2k
+    ;;
   s390 | s390x)
     CFLAGS="$CFLAGS -DCPU=\\\"s390\\\""
     supported_os="s390"
