--- azureus.spec~	2016-02-11 04:43:56.000000000 +0200
+++ azureus.spec	2016-02-11 04:53:06.110137584 +0200
@@ -118,12 +118,7 @@
 build-jar-repository -p build/libs bcprov apache-commons-cli log4j12-1.2.17 \
   junit apache-commons-lang json_simple
 
-#ppc seems to have eclipse-swt.ppc64 installed so libdir can't be used
-if [ -e /usr/lib/eclipse/swt.jar ];then
-  ln -s /usr/lib/eclipse/swt.jar build/libs
-else
-  ln -s /usr/lib64/eclipse/swt.jar build/libs
-fi
+ln -s %_jnidir/swt.jar build/libs
 
 ant jar
 
