--- batik.spec	2007-06-08 08:33:35 +0300
+++ /home/igor/src/RPM/SPECS/batik.spec	2007-06-08 08:37:15 +0300
@@ -353,16 +353,18 @@
 %endif
 
 %if ! %{bootstrap}
-%if %{gcj_support}
 %post squiggle
+%update_menus
+%if %{gcj_support}
 if [ -x %{_bindir}/rebuild-gcj-db ]
 then
   %{_bindir}/rebuild-gcj-db
 fi
 %endif
 
-%if %{gcj_support}
 %postun squiggle
+%update_menus
+%if %{gcj_support}
 if [ -x %{_bindir}/rebuild-gcj-db ]
 then
   %{_bindir}/rebuild-gcj-db
