diff -ru dynaop.0/src/dynaop/remote/PatchedObjectInputStream.java dynaop/src/dynaop/remote/PatchedObjectInputStream.java
--- dynaop.0/src/dynaop/remote/PatchedObjectInputStream.java	2004-01-05 17:12:54 +0000
+++ dynaop/src/dynaop/remote/PatchedObjectInputStream.java	2009-04-06 15:36:13 +0000
@@ -4,7 +4,7 @@
 import java.io.InputStream;
 import java.io.ObjectInputStream;
 
-import com.sun.corba.se.internal.io.ObjectStreamClass;
+import java.io.ObjectStreamClass;
 
 import dynaop.util.Classes;
 
