--- ./build/eclipse-R3_5_1-fetched-src/plugins/org.eclipse.swt/Eclipse SWT PI/common_j2se/org/eclipse/swt/internal/Library.java.old	2010-02-05 19:46:25 +0000
+++ ./build/eclipse-R3_5_1-fetched-src/plugins/org.eclipse.swt/Eclipse SWT PI/common_j2se/org/eclipse/swt/internal/Library.java	2010-02-05 19:53:56 +0000
@@ -215,6 +215,15 @@
 		if (mapName && load (path + SEPARATOR + mappedName2)) return;
 	}
 
+	/* Try eclipse syspath */
+	path = "/usr/lib64/eclipse";
+	if (load (path + SEPARATOR + mappedName1)) return;
+	if (mapName && load (path + SEPARATOR + mappedName2)) return;
+	path = "/usr/lib/eclipse";
+	if (load (path + SEPARATOR + mappedName1)) return;
+	if (mapName && load (path + SEPARATOR + mappedName2)) return;
+
+
 	/* Try loading library from java library path */
 	if (load (libName1)) return;
 	if (mapName && load (libName2)) return;
