--- build/eclipse-3.6.2-src/plugins/org.eclipse.core.net/natives/unix/linux/makefile	2010-03-29 14:36:30.000000000 +0000
+++ build/eclipse-3.6.2-src/plugins/org.eclipse.core.net/natives/unix/linux/makefile	2011-02-28 00:25:16.095307064 +0000
@@ -38,7 +38,7 @@
 link: compile
 	@echo "Building target: $(LIB_NAME_FULL)"
 	@echo "Invoking: GCC C Linker"
-	gcc $(LINKER_FLAGS) -shared -Wl,-soname,$(LIB_NAME) -o $(LIB_NAME_FULL) $(LIBS) $(GNOMEPROXY.O) -lc
+	gcc $(LINKER_FLAGS) -shared -Wl,-soname,$(LIB_NAME) -o $(LIB_NAME_FULL)  $(GNOMEPROXY.O) $(LIBS) -lc
 	@echo "Finished building target: $(LIB_NAME_FULL)"
 	@echo " "
 
