--- hadoop.spec~	2016-02-13 19:52:20.000000000 +0200
+++ hadoop.spec	2016-02-14 00:55:55.162574760 +0200
@@ -774,7 +774,7 @@
 # hdfs jar dependencies
 copy_dep_jars $basedir/share/%{name}/hdfs/lib %{buildroot}/%{_datadir}/%{name}/hdfs/lib
 %{_bindir}/xmvn-subst %{buildroot}/%{_datadir}/%{name}/hdfs/lib
-%{__ln_s} %{_javadir}/%{name}/%{name}-hdfs-bkjournal.jar %{buildroot}/%{_datadir}/%{name}/hdfs/lib
+%{__ln_s} %{_jnidir}/%{name}/%{name}-hdfs-bkjournal.jar %{buildroot}/%{_datadir}/%{name}/hdfs/lib
 pushd $basedir/share/%{name}/hdfs
   link_hadoop_jars %{buildroot}/%{_datadir}/%{name}/hdfs
 popd
