diff -ru hessian-3.0.8.old/src/com/caucho/burlap/client/BurlapProxy.java hessian-3.0.8/src/com/caucho/burlap/client/BurlapProxy.java
--- hessian-3.0.8.old/src/com/caucho/burlap/client/BurlapProxy.java	2004-04-14 07:51:54 +0000
+++ hessian-3.0.8/src/com/caucho/burlap/client/BurlapProxy.java	2008-09-21 18:00:02 +0000
@@ -94,10 +94,10 @@
     if (methodName.equals("equals") &&
         params.length == 1 && params[0].equals(Object.class)) {
       Object value = args[0];
-      if (value == null || ! Proxy.isProxyClass(value.getClass()))
+      if (value == null || ! java.lang.reflect.Proxy.isProxyClass(value.getClass()))
         return new Boolean(false);
 
-      BurlapProxy handler = (BurlapProxy) Proxy.getInvocationHandler(value);
+      BurlapProxy handler = (BurlapProxy) java.lang.reflect.Proxy.getInvocationHandler(value);
 
       return new Boolean(_url.equals(handler.getURL()));
     }
diff -ru hessian-3.0.8.old1/src/com/caucho/burlap/client/BurlapProxyFactory.java hessian-3.0.8/src/com/caucho/burlap/client/BurlapProxyFactory.java
--- hessian-3.0.8.old1/src/com/caucho/burlap/client/BurlapProxyFactory.java	2004-02-23 12:49:20 +0000
+++ hessian-3.0.8/src/com/caucho/burlap/client/BurlapProxyFactory.java	2008-09-21 18:03:53 +0000
@@ -202,7 +202,7 @@
   {
     BurlapProxy handler = new BurlapProxy(this, new URL(url));
 
-    return Proxy.newProxyInstance(api.getClassLoader(),
+    return java.lang.reflect.Proxy.newProxyInstance(api.getClassLoader(),
                                   new Class[] { api,
                                                 BurlapRemoteObject.class },
                                   handler);
diff -ru hessian-3.0.8.old1/src/com/caucho/hessian/client/HessianProxy.java hessian-3.0.8/src/com/caucho/hessian/client/HessianProxy.java
--- hessian-3.0.8.old1/src/com/caucho/hessian/client/HessianProxy.java	2004-05-12 15:15:32 +0000
+++ hessian-3.0.8/src/com/caucho/hessian/client/HessianProxy.java	2008-09-21 18:05:29 +0000
@@ -94,10 +94,10 @@
     if (methodName.equals("equals") &&
         params.length == 1 && params[0].equals(Object.class)) {
       Object value = args[0];
-      if (value == null || ! Proxy.isProxyClass(value.getClass()))
+      if (value == null || ! java.lang.reflect.Proxy.isProxyClass(value.getClass()))
         return new Boolean(false);
 
-      HessianProxy handler = (HessianProxy) Proxy.getInvocationHandler(value);
+      HessianProxy handler = (HessianProxy) java.lang.reflect.Proxy.getInvocationHandler(value);
 
       return new Boolean(_url.equals(handler.getURL()));
     }
diff -ru hessian-3.0.8.old1/src/com/caucho/hessian/client/HessianProxyFactory.java hessian-3.0.8/src/com/caucho/hessian/client/HessianProxyFactory.java
--- hessian-3.0.8.old1/src/com/caucho/hessian/client/HessianProxyFactory.java	2004-06-04 15:49:42 +0000
+++ hessian-3.0.8/src/com/caucho/hessian/client/HessianProxyFactory.java	2008-09-21 18:05:05 +0000
@@ -230,7 +230,7 @@
   {
     HessianProxy handler = new HessianProxy(this, new URL(url));
 
-    return Proxy.newProxyInstance(api.getClassLoader(),
+    return java.lang.reflect.Proxy.newProxyInstance(api.getClassLoader(),
 				  new Class[] { api,
                                                 HessianRemoteObject.class },
                                   handler);
