--- java-1.8.0-openjdk.spec	2020-10-09 16:50:06.920673923 +0300
+++ java-1.8.0-openjdk.spec	2020-10-09 16:53:59.222155598 +0300
@@ -1031,7 +1031,7 @@
 done
 
 # Setup nss.cfg
-sed -e "s:@NSS_LIBDIR@:%{NSS_LIBDIR}:g" %{SOURCE11} > nss.cfg
+#sed -e "s:@NSS_LIBDIR@:%{NSS_LIBDIR}:g" %{SOURCE11} > nss.cfg
 
 # Setup nss.fips.cfg
 sed -e "s:@NSS_LIBDIR@:%{NSS_LIBDIR}:g" %{SOURCE15} > nss.fips.cfg
@@ -1141,7 +1141,7 @@
 export JAVA_HOME=$(pwd)/%{buildoutputdir}/images/%{jdkimage}
 
 # Install nss.cfg right away as we will be using the JRE above
-install -m 644 nss.cfg $JAVA_HOME/jre/lib/security/
+#install -m 644 nss.cfg $JAVA_HOME/jre/lib/security/
 
 # Install nss.fips.cfg: NSS configuration for global FIPS mode (crypto-policies)
 install -m 644 nss.fips.cfg $JAVA_HOME/jre/lib/security/
@@ -1673,10 +1673,10 @@
 %{_mandir}/man1/tnameserv-%{uniquesuffix}.1*
 %{_mandir}/man1/unpack200-%{uniquesuffix}.1*
 %{_mandir}/man1/policytool-%{uniquesuffix}.1*
-%{_jvmdir}/%{jredir}/lib/security/nss.cfg
-%{_jvmdir}/%{jredir}/lib/security/nss.fips.cfg
-%config(noreplace) %{etcjavadir}/lib/security/nss.cfg
-%config(noreplace) %{etcjavadir}/lib/security/nss.fips.cfg
+#%{_jvmdir}/%{jredir}/lib/security/nss.cfg
+#%{_jvmdir}/%{jredir}/lib/security/nss.fips.cfg
+#%config(noreplace) %{etcjavadir}/lib/security/nss.cfg
+#%config(noreplace) %{etcjavadir}/lib/security/nss.fips.cfg
 %ifarch %{jit_arches}
 %ifnarch %{power64}
 %attr(444, root, root) %ghost %{_jvmdir}/%{jredir}/lib/%{archinstall}/server/classes.jsa
