--- Makefile.am.orig	2008-08-04 10:39:01 +0400
+++ Makefile.am	2008-08-04 10:41:36 +0400
@@ -919,10 +919,10 @@
 	if test "$(ENABLE_PLUGIN)" == "yes" ; \
 	then \
 	  $(CXX) $(CXXFLAGS) $(MOZILLA_CFLAGS) \
-	    $(MOZILLA_LIBS) $(GLIB_CFLAGS) $(GLIB_LIBS) \
-	    $(GTK_CFLAGS) $(GTK_LIBS) \
+	    $(GLIB_CFLAGS) $(GTK_CFLAGS) \
 	    -DPACKAGE_VERSION="\"$(PACKAGE_VERSION)\"" \
-	    -fpic -shared -o $@ $< ; \
+	    -fpic -shared -o $@ $<  \
+	    $(MOZILLA_LIBS) $(GLIB_LIBS) $(GTK_LIBS); \
 	fi
 
 clean-gcjwebplugin:
