--- java-1.8.0-openjdk.spec.orig	2016-10-20 10:36:54.410699307 +0300
+++ java-1.8.0-openjdk.spec	2016-10-20 10:41:40.689688590 +0300
@@ -724,7 +724,7 @@
     OUTPUT_FILE=$NAME$suffix.$EXT
     sed -e s:#JAVA_HOME#:%{sdkbindir}:g $file > $OUTPUT_FILE
     sed -i -e  s:#JRE_HOME#:%{jrebindir}:g $OUTPUT_FILE
-    sed -i -e  s:#ARCH#:%{version}-%{release}.%{_arch}$suffix:g $OUTPUT_FILE
+    sed -i -e  s:#ARCH#:%{_arch}$suffix:g $OUTPUT_FILE
 done
 done
 sed -i -e 's,DEF_OBJCOPY=/usr/bin/objcopy,DEF_OBJCOPY=/usr/bin/NO-objcopy,' jdk8/hotspot/make/linux/makefiles/defs.make
