--- javapackages/java-utils/java-functions	2018-08-06 18:09:06.000000000 +0300
+++ javapackages/java-utils/java-functions	2021-05-10 19:50:14.455480922 +0300
@@ -250,7 +250,7 @@
 }
 
 # Run application
-run()
+jvm_run()
 {
     set_javacmd
 
@@ -266,6 +266,10 @@
     # let's start
     eval "exec \"\${JAVACMD}\" ${FLAGS} -classpath \"\${CLASSPATH}\" ${OPTIONS} \"\${MAIN_CLASS}\" \"\${@}\""
 }
+run()
+{
+    jvm_run "${@}"
+}
 
 # Set JVM-related directories
 set_jvm_dirs()
--- javapackages/macros.d/macros.jpackage	2018-08-06 18:09:06.000000000 +0300
+++ javapackages/macros.d/macros.jpackage	2021-05-10 19:47:24.332625409 +0300
@@ -37,7 +37,7 @@
 %jpackage_script() \
 install -d -m 755 %{buildroot}%{_bindir} \
 cat > %{buildroot}%{_bindir}/%5 << EOF \
-#!/bin/bash\
+#!/bin/sh\
 #\
 # %{name} script\
 # JPackage Project <http://www.jpackage.org/>\
@@ -47,8 +47,9 @@
 . @{javadir}-utils/java-functions\
 \
 # Source system prefs\
-if [ -f %{_sysconfdir}/java/%{name}.conf ] ; then\
-  . %{_sysconfdir}/java/%{name}.conf\
+_sysconf_file=%{_sysconfdir}/java/%{name}.conf\
+if [ -f \\\$_sysconf_file ] ; then\
+  . \\\$_sysconf_file\
 fi\
 \
 # Source user prefs\
@@ -69,7 +70,7 @@
 set_options \\$BASE_OPTIONS\
 \
 # Let's start\
-run "\\$@"\
+jvm_run "\\$@"\
 EOF\
 chmod 755 %{buildroot}%{_bindir}/%5\
 %{nil}
