--- javapackages-5.0.0/macros.d/macros.fjava	2017-06-21 12:04:13.009318912 +0300
+++ javapackages-5.0.0/macros.d/macros.fjava	2017-11-03 18:55:14.325423624 +0200
@@ -195,8 +195,6 @@
 
 %add_maven_depmap(f:a:v:) \
 set -e \
-%{lua:io.stderr:write("[WARNING] Deprecated \%add_maven_depmap macro is used. ", \
-"Please use \%mvn_artifact and %mvn_install instead.\\n")} \
 # default values \
 %if %# == 0 \
    _pompart="JPP-%{?scl:%{pkg_name}}%{!?scl:%{name}}.pom" \
@@ -268,8 +268,7 @@
 # directory containing pom.xml. If POM location is not given then pom.xml from
 # current working directory is used.
 #
-%pom_remove_dep(rf)     %{expand: %{__pom_call}remove_dep %**
-}
+%pom_remove_dep     %{__pom_call}remove_dep 
 
 
 # %pom_remove_plugin - remove Maven plugin invocation from POM file
@@ -287,8 +286,7 @@
 # directory containing pom.xml. If POM location is not given then pom.xml from
 # current working directory is used.
 #
-%pom_remove_plugin(rf)  %{expand: %{__pom_call}remove_plugin  %**
-}
+%pom_remove_plugin  %{__pom_call}remove_plugin 
 
 
 # %pom_disable_module - disable given project module in POM file
@@ -305,8 +303,7 @@
 # directory containing pom.xml. If POM location is not given then pom.xml from
 # current working directory is used.
 #
-%pom_disable_module() %{expand: %{__pom_call}disable_module %*
-}
+%pom_disable_module %{__pom_call}disable_module 
 
 
 # %pom_xpath_remove - remove an XML node from POM file
@@ -330,11 +327,9 @@
 # because it selects nodes from pom namespace, but "environment/os" won't find
 # anything because it looks for nodes that don't belong to any XML namespace.
 #
-%pom_xpath_remove(rf)   %{expand: %{__pom_call}xpath_remove   %**
-}
+%pom_xpath_remove   %{__pom_call}xpath_remove 
 
-%pom_xpath_disable()   %{expand: %{__pom_call}xpath_disable   %**
-}
+%pom_xpath_disable   %{__pom_call}xpath_disable 
 
 # %pom_xpath_inject - inject XML code into POM file
 #
@@ -351,8 +346,7 @@
 # directory containing pom.xml. If POM location is not given then pom.xml from
 # current working directory is used.
 #
-%pom_xpath_inject()   %{expand: %{__pom_call}xpath_inject   %*
-}
+%pom_xpath_inject   %{__pom_call}xpath_inject 
 
 
 # %pom_xpath_replace - replace XML node from POM file with given XML code
@@ -370,8 +364,7 @@
 # directory containing pom.xml. If POM location is not given then pom.xml from
 # current working directory is used.
 #
-%pom_xpath_replace(rf)  %{expand: %{__pom_call}xpath_replace  %**
-}
+%pom_xpath_replace  %{__pom_call}xpath_replace  
 
 
 # %pom_xpath_set - set text contents of XML node from POM file
@@ -389,8 +382,7 @@
 # directory containing pom.xml. If POM location is not given then pom.xml from
 # current working directory is used.
 #
-%pom_xpath_set(rf)      %{expand: %{__pom_call}xpath_set      %**
-}
+%pom_xpath_set      %{__pom_call}xpath_set  
 
 
 # %pom_add_parent - add parent POM reference to POM file
@@ -408,8 +400,7 @@
 # directory containing pom.xml. If POM location is not given then pom.xml from
 # current working directory is used.
 #
-%pom_add_parent()     %{expand: %{__pom_call}add_parent     %*
-}
+%pom_add_parent     %{__pom_call}add_parent    
 
 
 # %pom_remove_parent - remove parent POM reference from POM file
@@ -422,8 +413,7 @@
 # directory containing pom.xml. If POM location is not given then pom.xml from
 # current working directory is used.
 #
-%pom_remove_parent()  %{expand: %{__pom_call}remove_parent  %*
-}
+%pom_remove_parent  %{__pom_call}remove_parent  
 
 
 # %pom_set_parent - set parent POM reference in POM file
@@ -438,8 +428,7 @@
 # invocation of %pom_add_parent. See documentation for these macros for more
 # information about parameters.
 #
-%pom_set_parent()     %{expand: %{__pom_call}set_parent     %*
-}
+%pom_set_parent     %{__pom_call}set_parent  
 
 
 # %pom_add_dep[_mgmt] - add dependency to POM file
@@ -466,10 +455,8 @@
 # Extra XML is additional XML code to be added into the <dependency> node. This
 # parameter can be ommited, in which case no extra XML is appended.
 #
-%pom_add_dep()        %{expand: %{__pom_call}add_dep        %*
-}
-%pom_add_dep_mgmt()   %{expand: %{__pom_call}add_dep_mgmt   %*
-}
+%pom_add_dep        %{__pom_call}add_dep  
+%pom_add_dep_mgmt   %{__pom_call}add_dep_mgmt 
 
 
 # %pom_add_plugin - add plugin invocation to POM file
@@ -492,11 +479,9 @@
 # Extra XML is additional XML code to be added into the <plugin> node. This
 # parameter can be ommited, in which case no extra XML is appended.
 #
-%pom_add_plugin()     %{expand: %{__pom_call}add_plugin     %*
-}
+%pom_add_plugin     %{__pom_call}add_plugin 
 
-%pom_change_dep(rf)     %{expand: %{__pom_call}change_dep    %**
-}
+%pom_change_dep     %{__pom_call}change_dep
 
 # %javadoc_package - generate javadoc subpackage definition
 #
