--- macros.d/macros.jpackage	2016-01-14 21:17:59.000000000 +0200
+++ macros.d/macros.jpackage	2016-01-14 21:37:25.000000000 +0200
@@ -86,7 +86,7 @@
 #   - jars for Java standard x.y.z (usually symlinks to  %{_jnidir}-ext)
 # To simplify things only %{_jnidir} is defined.
 #
-%_jnidir        %{_prefix}/lib/java
+%_jnidir        %{_libdir}/java
 
 #
 # Root directory where all javadoc is installed. Also already in RH macros.
@@ -132,6 +132,56 @@
 %javadoc        %{java_home}/bin/javadoc
 
 #==============================================================================
+#==============================================================================
+#
+# add_to_depmap adds an entry to the depmap. The arguments are:
+#
+# %1 the original groupid
+# %2 the original artifact id
+# %3 the version
+# %4 the new groupid
+# %5 the new artifactid
+#
+
+%add_to_maven_depmap() \
+install -dm 755 $RPM_BUILD_ROOT/%{_mavendepmapfragdir}\
+cat >>$RPM_BUILD_ROOT/%{_mavendepmapfragdir}/%{name}<< EOF\
+<dependency>\
+    <maven>\
+        <groupId>%1</groupId>\
+        <artifactId>%2</artifactId>\
+        <version>%3</version>\
+    </maven>\
+    <jpp>\
+        <groupId>%4</groupId>\
+        <artifactId>%5</artifactId>\
+        <version>%3</version>\
+    </jpp>\
+</dependency>\
+\
+EOF\
+%{nil}
+
+%add_to_maven_depmap_at() \
+install -dm 755 $RPM_BUILD_ROOT/%{_mavendepmapfragdir}\
+cat >>$RPM_BUILD_ROOT/%{_mavendepmapfragdir}/%1<< EOF\
+<dependency>\
+    <maven>\
+        <groupId>%2</groupId>\
+        <artifactId>%3</artifactId>\
+        <version>%4</version>\
+    </maven>\
+    <jpp>\
+        <groupId>%5</groupId>\
+        <artifactId>%6</artifactId>\
+        <version>%4</version>\
+    </jpp>\
+</dependency>\
+\
+EOF\
+%{nil}
+
+#==============================================================================
 # ---- Java extension handling macros
 
 #
@@ -167,7 +217,7 @@
 %jpackage_script() \
 install -d -m 755 %{buildroot}%{_bindir} \
 cat > %{buildroot}%{_bindir}/%5 << EOF \
-#!/bin/bash\
+#!/bin/sh\
 #\
 # %{name} script\
 # JPackage Project <http://www.jpackage.org/>\
