--- maven2-plugins/maven-assembly-plugin/src/main/java/org/apache/maven/plugin/assembly/utils/PropertiesInterpolationValueSource.java	2006-08-03 02:45:53.000000000 +0000
+++ maven2-plugins/maven-assembly-plugin/src/main/java/org/apache/maven/plugin/assembly/utils/PropertiesInterpolationValueSource.java	2011-01-26 14:52:28.586940473 +0000
@@ -3,6 +3,7 @@
 import org.codehaus.plexus.util.interpolation.ValueSource;
 
 import java.util.Properties;
+import java.util.List;
 
 
 public class PropertiesInterpolationValueSource
@@ -21,4 +22,11 @@
         return properties.getProperty( key );
     }
 
+    public List getFeedback()
+    {
+	return new java.util.ArrayList();
+    }
+
+    public void clearFeedback() {}
+
 }
--- maven2-plugins/maven-war-plugin/src/main/java/org/apache/maven/plugin/war/MappingUtils.java	2011-01-26 15:06:06.470898432 +0000
+++ maven2-plugins/maven-war-plugin/src/main/java/org/apache/maven/plugin/war/MappingUtils.java	2011-01-26 15:22:52.778934116 +0000
@@ -95,6 +95,13 @@
             return properties.getProperty( key );
         }
 
+	public List getFeedback()
+	{
+	    return new java.util.ArrayList();
+	}
+
+	public void clearFeedback() {}
+
     }
 
 
