Index: maven2/maven2/maven-artifact-ant/src/main/java/org/apache/maven/artifact/ant/AbstractArtifactTask.java
===================================================================
--- maven2/maven2/maven-artifact-ant/src/main/java/org/apache/maven/artifact/ant/AbstractArtifactTask.java	(revision 541989)
+++ maven2/maven2/maven-artifact-ant/src/main/java/org/apache/maven/artifact/ant/AbstractArtifactTask.java	(working copy)
@@ -88,7 +88,7 @@
         ArtifactRepositoryLayout repositoryLayout =
             (ArtifactRepositoryLayout) lookup( ArtifactRepositoryLayout.ROLE, localRepository.getLayout() );
 
-        return new DefaultArtifactRepository( "local", "file://" + localRepository.getLocation(), repositoryLayout );
+        return new DefaultArtifactRepository( "local", "file://" + localRepository.getPath(), repositoryLayout );
     }
 
     protected ArtifactRepository createRemoteArtifactRepository( RemoteRepository repository )
@@ -169,7 +169,7 @@
         Settings settings = getSettings();
         LocalRepository localRepository = new LocalRepository();
         localRepository.setId( "local" );
-        localRepository.setLocation( new File( settings.getLocalRepository() ) );
+        localRepository.setPath( new File( settings.getLocalRepository() ) );
         return localRepository;
     }
 
Index: maven2/maven2/maven-artifact-ant/src/main/java/org/apache/maven/artifact/ant/DependenciesTask.java
===================================================================
--- maven2/maven2/maven-artifact-ant/src/main/java/org/apache/maven/artifact/ant/DependenciesTask.java	(revision 541989)
+++ maven2/maven2/maven-artifact-ant/src/main/java/org/apache/maven/artifact/ant/DependenciesTask.java	(working copy)
@@ -201,14 +201,14 @@
         }
 
         FileList fileList = new FileList();
-        fileList.setDir( getLocalRepository().getLocation() );
+        fileList.setDir( getLocalRepository().getPath() );
 
         FileSet fileSet = new FileSet();
         fileSet.setProject( getProject() );
         fileSet.setDir( fileList.getDir( getProject() ) );
 
         FileList sourcesFileList = new FileList();
-        sourcesFileList.setDir( getLocalRepository().getLocation() );
+        sourcesFileList.setDir( getLocalRepository().getPath() );
 
         FileSet sourcesFileSet = new FileSet();
         sourcesFileSet.setDir( sourcesFileList.getDir( getProject() ) );
Index: maven2/maven2/maven-artifact-ant/src/main/java/org/apache/maven/artifact/ant/LocalRepository.java
===================================================================
--- maven2/maven2/maven-artifact-ant/src/main/java/org/apache/maven/artifact/ant/LocalRepository.java	(revision 541989)
+++ maven2/maven2/maven-artifact-ant/src/main/java/org/apache/maven/artifact/ant/LocalRepository.java	(working copy)
@@ -30,15 +30,15 @@
 public class LocalRepository
     extends Repository
 {
-    private File location;
+    private File path;
 
-    public File getLocation()
+    public File getPath()
     {
-        return ( (LocalRepository) getInstance() ).location;
+        return ( (LocalRepository) getInstance() ).path;
     }
 
-    public void setLocation( File location )
+    public void setPath( File path )
     {
-        this.location = location;
+        this.path = path;
     }
 }
Index: sample.build.xml
===================================================================
--- maven2/maven2/maven-artifact-ant/sample.build.xml	(revision 541989)
+++ maven2/maven2/maven-artifact-ant/sample.build.xml	(working copy)
@@ -30,7 +30,7 @@
       <classpath refid="maven.classpath"/>
     </typedef>
 
-    <artifact:localRepository id="local.repository" location="${basedir}/target/local-repo" layout="default"/>
+    <artifact:localRepository id="local.repository" path="${basedir}/target/local-repo" layout="default"/>
 
     <artifact:remoteRepository id="deploy.repository" url="file://localhost/${basedir}/target/deployment-repo" layout="default"/>
 
@@ -63,7 +63,7 @@
   <target name="test-legacy-pom" depends="initTaskDefs">
     <artifact:dependencies>
       <dependency groupId="geronimo-spec" artifactId="geronimo-spec-javamail" version="1.3.1-rc5" />
-      <localRepository location="${basedir}/target/local-repo-legacy" />
+      <localRepository path="${basedir}/target/local-repo-legacy" />
       <remoteRepository url="http://repo1.maven.org/maven" layout="legacy" id="central" />
     </artifact:dependencies>
   </target>
@@ -168,7 +168,7 @@
       <arg line="create ${repo.path.unix}"/>
     </exec>
 
-    <artifact:localRepository id="local.repository" location="${basedir}/target/local-repo" layout="default"/>
+    <artifact:localRepository id="local.repository" path="${basedir}/target/local-repo" layout="default"/>
 
     <artifact:remoteRepository id="deploy.repository" url="scm:svn:${scm.url}" layout="default"/>
 
@@ -189,7 +189,7 @@
       <arg line="co ${scm.url} -N local-repo-scm"/>
     </exec>
 
-    <artifact:localRepository id="local.repository.scm" location="${basedir}/target/local-repo-scm" layout="default"/>
+    <artifact:localRepository id="local.repository.scm" path="${basedir}/target/local-repo-scm" layout="default"/>
 
     <artifact:dependencies pathId="dependency.classpath.scm">
       <dependency groupId="org.apache.maven.wagon" artifactId="wagon-provider-test" version="1.0-alpha-4"/>
