diff -ru maven2/maven2/maven-project/src/main/java/org/apache/maven/project/DefaultMavenProjectBuilder.java maven2/maven2/maven-project/src/main/java/org/apache/maven/project/DefaultMavenProjectBuilder.java
--- maven2.orig//maven2/maven-project/src/main/java/org/apache/maven/project/DefaultMavenProjectBuilder.java	2007-10-16 21:43:07.000000000 +0000
+++ maven2/maven2/maven-project/src/main/java/org/apache/maven/project/DefaultMavenProjectBuilder.java	2010-09-24 17:32:17.080127109 +0000
@@ -1028,21 +1028,6 @@
         Model originalModel = ModelUtils.cloneModel( model );
 
 
-        if ( !model.getRepositories().isEmpty() )
-        {
-            List respositories = buildArtifactRepositories( model );
-
-            for ( Iterator it = respositories.iterator(); it.hasNext(); )
-            {
-                ArtifactRepository repository = (ArtifactRepository) it.next();
-
-                if ( !aggregatedRemoteWagonRepositories.contains( repository ) )
-                {
-                    aggregatedRemoteWagonRepositories.add( repository );
-                }
-            }
-        }
-
         //TODO mkleint - use the (Container, Properties constructor to make system properties embeddable
         ProfileManager profileManager;
         if ( externalProfileManager != null && externalProfileManager instanceof DefaultProfileManager )
@@ -1080,6 +1065,21 @@
                 e.getMessage(), e );
         }
 
+        if ( !model.getRepositories().isEmpty() )
+        {
+            List respositories = buildArtifactRepositories( model );
+
+            for ( Iterator it = respositories.iterator(); it.hasNext(); )
+            {
+                ArtifactRepository repository = (ArtifactRepository) it.next();
+
+                if ( !aggregatedRemoteWagonRepositories.contains( repository ) )
+                {
+                    aggregatedRemoteWagonRepositories.add( repository );
+                }
+            }
+        }
+
         MavenProject project = new MavenProject( model );
 
         project.setActiveProfiles( activeProfiles );
