diff -ur mojo-maven2-plugins-17/webstart/webstart-maven-plugin/src/main/java/org/codehaus/mojo/webstart/JarUnsignMojo.java mojo-maven2-plugins-17/webstart/webstart-maven-plugin/src/main/java/org/codehaus/mojo/webstart/JarUnsignMojo.java
--- mojo-maven2-plugins-17.0/webstart/webstart-maven-plugin/src/main/java/org/codehaus/mojo/webstart/JarUnsignMojo.java	2008-03-28 12:22:57.000000000 +0000
+++ mojo-maven2-plugins-17/webstart/webstart-maven-plugin/src/main/java/org/codehaus/mojo/webstart/JarUnsignMojo.java	2011-01-08 13:05:21.382390199 +0000
@@ -148,6 +148,8 @@
             unArchiver.setSourceFile( jarFile );
             unArchiver.setDestDirectory( tempDir );
             unArchiver.extract();            
+        } catch ( java.io.IOException ex)  {
+            throw new MojoExecutionException( "IO Error unpacking file: " + jarFile + "to: " + tempDir, ex );            
         } catch ( ArchiverException ex)  {
             throw new MojoExecutionException( "Error unpacking file: " + jarFile + "to: " + tempDir, ex );            
         } catch ( NoSuchArchiverException ex ) {
