--- sdljava-0.9.1/src/org/gljava/opengl/native/ftgl/post-process.rb	2005-02-20 22:19:13.000000000 +0000
+++ sdljava-0.9.1/src/org/gljava/opengl/native/ftgl/post-process.rb	2011-09-03 11:44:43.383846402 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/ruby -w
 
 require "find"
-require "ftools"
+require "fileutils"
 
 # temporary hack until I can figure out a better way to do this
 #    process the given files, change the methods names to start
@@ -110,7 +110,7 @@
   f.close
 
   # move it to the actual file now
-  File.move("#{path}.tmp", path)
+  FileUtils.mv("#{path}.tmp", path)
 end
 
 
