diff --git a/patches/tomcat5-5.5.init b/patches/tomcat5-5.5.init
index f6e502b..24da8be 100644
--- a/patches/tomcat5-5.5.init
+++ b/patches/tomcat5-5.5.init
@@ -32,9 +32,9 @@ fi
 
 # For SELinux we need to use 'runuser' not 'su'
 if [ -x "/sbin/runuser" ]; then
-    SU="/sbin/runuser"
+    SU="/sbin/runuser -s /bin/sh"
 else
-    SU="su"
+    SU="su -s /bin/sh"
 fi
 
 # Get the tomcat config (use this for environment specific settings)
@@ -174,7 +174,7 @@ function start() {
             ${CATALINA_HOME}/conf/server.xml
     fi
     $TOMCAT_RELINK_SCRIPT
-    $SU - $TOMCAT_USER -c "$TOMCAT_SCRIPT start" >> $TOMCAT_LOG 2>&1
+    $SU $TOMCAT_USER -c "$TOMCAT_SCRIPT start" >> $TOMCAT_LOG 2>&1
     RETVAL="$?"
     if [ "$RETVAL" -eq 0 ]; then 
         echo_success
@@ -217,7 +217,7 @@ function stop() {
 	echo_success
     fi
     if [ -f "/var/lock/subsys/${NAME}" ]; then
-        $SU - $TOMCAT_USER -c "$TOMCAT_SCRIPT stop" >> $TOMCAT_LOG 2>&1
+        $SU $TOMCAT_USER -c "$TOMCAT_SCRIPT stop" >> $TOMCAT_LOG 2>&1
         RETVAL="$?"
         if [ "$RETVAL" -eq "0" ]; then
             count="0"
