--- a/icedtea-web/launchers/xdesktop.in/itweb-settings.desktop.in
+++ b/icedtea-web/launchers/xdesktop.in/itweb-settings.desktop.in
@@ -3,11 +3,14 @@ Name=IcedTea-Web Control Panel
 Name[de]=IcedTea-Web Systemsteuerung
 Name[pl]=Panel sterowania IcedTea-Web
 Name[cs]=Ovládací panel IcedTea-Web
+Name[ru]=Панель управления IcedTea-Web
 GenericName=Control Panel
+GenericName[ru]=Панель управления
 Comment=Configure IcedTea-Web (javaws and plugin)
 Comment[de]=Konfiguriert IcedTea-Web (javaws und Plug-in)
 Comment[pl]=Konfiguruj IcedTea-Web (javaws i wtyczkę)
 Comment[cs]=Konfigurace aplikace IcedTea-Web (javaws a zásuvný modul)
+Comment[ru]=Настройка IcedTea-Web (javaws и подключаемый модуль)
 Exec=PATH_TO_ITWEB_SETTINGS
 Icon=javaws
 Terminal=false
--- a/icedtea-web/launchers/xdesktop.in/javaws.desktop.in
+++ b/icedtea-web/launchers/xdesktop.in/javaws.desktop.in
@@ -2,6 +2,7 @@
 Name=IcedTea Web Start
 GenericName=Java Web Start
 Comment=IcedTea Application Launcher
+Comment[ru]=Средство запуска приложения IcedTea
 Exec=PATH_TO_JAVAWS %u
 Icon=javaws
 Terminal=false
--- a/icedtea-web/launchers/xdesktop.in/policyeditor.desktop.in
+++ b/icedtea-web/launchers/xdesktop.in/policyeditor.desktop.in
@@ -1,7 +1,10 @@
 [Desktop Entry]
 Name=IcedTea-Web Policy Editor
-GenericName=Policy Tool 
+Name[ru]=Редактор политики IcedTea-Web
+GenericName=Policy Tool
+GenericName[ru]=Средство управления политикой
 Comment=Edit Java Applet policy and permission settings
+Comment[ru]=Редактирование политики и параметров разрешений приложений Java
 Exec=PATH_TO_POLICYEDITOR
 Icon=javaws
 Terminal=false
