diff -ru clirr-0.6.old/core/src/java/net/sf/clirr/core/internal/bcel/BcelJavaType.java clirr-0.6/core/src/java/net/sf/clirr/core/internal/bcel/BcelJavaType.java
--- clirr-0.6.old/core/src/java/net/sf/clirr/core/internal/bcel/BcelJavaType.java	2009-04-06 09:27:49 +0000
+++ clirr-0.6/core/src/java/net/sf/clirr/core/internal/bcel/BcelJavaType.java	2009-04-06 10:01:47 +0000
@@ -66,12 +66,8 @@
         {
             return new JavaType[0];
         }
-        try {
             final JavaClass[] superClasses = clazz.getSuperClasses();
             return convertToJavaTypeArray(superClasses);
-        } catch (ClassNotFoundException cnfe) {
-            return null;
-        }
     }
 
     public JavaType[] getAllInterfaces() {
@@ -79,12 +75,8 @@
         {
             return new JavaType[0];
         }
-        try {
             final JavaClass[] interfaces = clazz.getAllInterfaces();
             return convertToJavaTypeArray(interfaces);
-        } catch (ClassNotFoundException cnfe) {
-            return null;
-        }
     }
 
     public JavaType[] getInnerClasses() {
